File: src/test/java/com/example/pages/WindowsPage.java
package com.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

import java.util.Set;

public class WindowsPage extends SeleniumBasePage {

    private final By openWindowBtn = By.id("open-window-btn");

    public WindowsPage(WebDriver driver) {
        super(driver);
    }

    public String switchToNewWindow() {
        String original = driver.getWindowHandle();
        click(openWindowBtn);

        Set<String> handles = driver.getWindowHandles();
        for (String h : handles) {
            if (!h.equals(original)) {
                driver.switchTo().window(h);
                return driver.getTitle();
            }
        }
        return null;
    }
}

File: python/pages/windows_page.py
from selenium.webdriver.common.by import By
from .base_page import BasePage


class WindowsPage(BasePage):

    OPEN_BTN = (By.ID, "open-window-btn")

    def switch_to_new_window(self):
        original = self.driver.current_window_handle

        self.click(self.OPEN_BTN)

        for handle in self.driver.window_handles:
            if handle != original:
                self.driver.switch_to.window(handle)
                return self.driver.title

        return None
